src/node/stage/joining.rs

```diff
@@ -33,47 +32,104 @@ pub(crate) struct Joining {
     section_key: bls::PublicKey,
     // Whether we are joining as infant or relocating.
     join_type: JoinType,
-    // Token for the join request timeout.
-    timer_token: u64,
-    // Message we can't handle until we are joined.
-    msg_backlog: Vec<QueuedMessage>,
+    comm: Comm,
+    node_info: NodeInfo,
 }
 
 impl Joining {
-    pub fn new(
-        core: &mut Core,
+    pub async fn new(
+        comm: Comm,
         elders_info: EldersInfo,
         section_key: bls::PublicKey,
         relocate_payload: Option<RelocatePayload>,
-        msg_backlog: Vec<QueuedMessage>,
```

Comment:
Why was this removed?

Reply:
Basically to not cache/accumulate anything, if a message arrives and it was not for us we should ignore, if it arrived in a different order than expected we should be able to process it anyways, so there shouldn't be a need to keep messages in a backlog/cache for future processing. E.g. If we are bootstrapping still, and we receive a message which we'll need when we are in a diff stage, well it will need to be resent, too bad it was sent earlier which may be weird or a bug as well.

Reply:
So on one hand, we should never have a situation where we absolutely need some message to be processed while also having the message being sent only once. So it shouldn't be necessary to backlog anything, because any missed message will be sent again. On the other hand, the backlog is an optimization to avoid the resends which can make things faster and involving less traffic. 

But OK, maybe we should first make sure things work even without the backlog, and possibly add it later purely as an optimization. Let's remove it then.